subsys/bluetooth/mesh/adv_ext.c

```diff
@@ -51,62 +42,133 @@ enum {
 	ADV_FLAGS_NUM
 };
 
-static struct {
+typedef struct net_buf* (*adv_buf_handler_t)(k_timeout_t timeout);
```

Comment:
```suggestion
typedef struct net_buf* (*adv_buf_alloc_t)(k_timeout_t timeout);
```
with a corresponding name change in the `ext_adv` struct below.

This could also be handled by replacing the two separate tag specific adv_get and buf_ready functions with a single one that takes LOCAL/RELAY tags as a bitfield. Could even add a third GATT_PROXY tag, and all the if/else business surrounding the allocation of advertising instances depending on configuration would go away, and the buffer selection would just be determined by what tags an advertiser instance has. 

There's currently a lot of `#if/#else` code here that doesn't have to be that separated if we accessed allocation buffers based on parameters, rather than functions.

This could also be achieved by adding "role" flags to the existing `ext_adv` `flags` field. The logic for assign the flags could be done in the init function to keep it simple. I would argue that you wouldn't even need three different sets of advertiser instances, just make one pool, and assign the "role" tags to the appropriate ones, depending on the configuration.

Reply:
>  I would argue that you wouldn't even need three different sets of advertiser instances, just make one pool, and assign the "role" tags to the appropriate ones, depending on the configuration.

https://github.com/zephyrproject-rtos/zephyr/pull/40031#discussion_r750300086

Add section for this.